pkg/exchange/ftx/stream.go

```diff
@@ -17,36 +17,48 @@ const endpoint = "wss://ftx.com/ws/"
 type Stream struct {
 	*types.StandardStream
 
-	ws *service.WebsocketClientBase
+	ws           *service.WebsocketClientBase
+	klineMessage chan types.KLine
+	exchange     *Exchange
+	ctx          context.Context
+	isConnected  bool
 
 	// publicOnly can only be configured before connecting
 	publicOnly int32
 
-	key    string
-	secret string
+	key        string
+	secret     string
+	subAccount string
 
 	// subscriptions are only accessed in single goroutine environment, so I don't use mutex to protect them
 	subscriptions []websocketRequest
 }
 
-func NewStream(key, secret string) *Stream {
+func NewStream(key, secret string, subAccount string, e *Exchange) *Stream {
 	s := &Stream{
+		exchange:       e,
+		isConnected:    false,
 		key:            key,
+		klineMessage:   make(chan types.KLine),
 		secret:         secret,
+		subAccount:     subAccount,
 		StandardStream: &types.StandardStream{},
 		ws:             service.NewWebsocketClientBase(endpoint, 3*time.Second),
 	}
 
 	s.ws.OnMessage((&messageHandler{StandardStream: s.StandardStream}).handleMessage)
 	s.ws.OnConnected(func(conn *websocket.Conn) {
-		subs := []websocketRequest{newLoginRequest(s.key, s.secret, time.Now())}
+		s.EmitConnect()
+
+		subs := []websocketRequest{newLoginRequest(s.key, s.secret, time.Now(), s.subAccount)}
 		subs = append(subs, s.subscriptions...)
 		for _, sub := range subs {
 			if err := conn.WriteJSON(sub); err != nil {
 				s.ws.EmitError(fmt.Errorf("failed to send subscription: %+v", sub))
 			}
 		}
 	})
+	go s.handleChannelKlineMessage()
```

Comment:
we should start the go routine when we call Connect, not in the constructor.

Reply:
fixed